src/libs/auth.ts

```diff
@@ -109,59 +149,146 @@ const getSigninArgs = (includeBillingScope) => {
-  const jwtExpiresAt: number = (decodedJWT as any).exp; // time in seconds when the JWT expires, after which the JWT should not be read from
+export const loadAuthToken = async (includeBillingScope = false, popUp = false): Promise<User | null | boolean> => {
+  // here we should update the auth token id
+  // might want to take refresh token and encrypt it to compare to a saved encrypted token
+  // to see if they are the same => track refresh update
+  // track lifespan of current authToken
+  // need to send information about the last authToken
+  const oldAuthTokenMetadata = authStore.get().authTokenMetadata; // this can be null (first token), so cover that case
 
   authStore.update((state) => ({
     ...state,
-    hasGcpBillingScopeThroughB2C: includeBillingScope,
-    sessionId,
-    sessionStartTime,
     authTokenMetadata: {
-      createdAt: authTokenCreatedAt,
-      expiresAt: authTokenExpiresAt,
+      ...oldAuthTokenMetadata,
+      totalAuthTokenLoadAttemptsThisSession:
+        authStore.get().authTokenMetadata.totalAuthTokenLoadAttemptsThisSession + 1,
     },
   }));
-  Ajax().Metrics.captureEvent(Events.userLogin, {
-    authProvider: user.profile.idp,
-    sessionStartTime: Utils.makeCompleteDate(sessionStartTime),
-    authTokenCreatedAt: Utils.formatTimestampInSeconds(authTokenCreatedAt),
-    authTokenExpiresAt: Utils.formatTimestampInSeconds(authTokenExpiresAt),
-    jwtExpiresAt: Utils.formatTimestampInSeconds(jwtExpiresAt),
-  });
-  return user;
+
+  const reloadedAuthTokenState: User | boolean | null = popUp
+    ? await tryLoadAuthTokenPopUp(includeBillingScope)
+    : await tryLoadAuthTokenSilent(includeBillingScope);
+
+  if (reloadedAuthTokenState instanceof User) {
+    const user: User = reloadedAuthTokenState as User;
+    const userJWT: string = user.id_token!;
+    const decodedJWT: JwtPayload = jwtDecode<JwtPayload>(userJWT);
+    const authTokenCreatedAt: number = (decodedJWT as any).auth_time; // time in seconds when authorization token was created
+    const authTokenExpiresAt: number = user.expires_at!; // time in seconds when authorization token expires, as given by the oidc client
+    const authTokenId: string = uuid();
+    const jwtExpiresAt: number = (decodedJWT as any).exp; // time in seconds when the JWT expires, after which the JWT should not be read from
+    // refresh token can be stored and labeled with a uuid, then only log the uuid
+    const refreshToken: string | undefined = user.refresh_token;
+    const isNewRefreshToken = refreshToken !== null && refreshToken !== oldAuthTokenMetadata.refreshToken;
+    if (isNewRefreshToken) {
+      authStore.update((state) => ({
+        ...state,
+        refreshTokenMetadata: {
+          id: uuid(),
+          token: refreshToken,
+          createdAt: authTokenCreatedAt,
+          // I do not think we have any way of getting this info except by looking at B2C configs
```

Comment:
What info is this comment referring to?

Reply:
The `refreshTokenExpiresAt` time. Since we cannot seem to decode the `refresh_token`, we are making the assumption that the token lasts for 24 hours which is standard for single page apps.

Reply:
Ah, I see. In that case, I would suggest rewording this to something like:

```suggestion
          // Token expiration is set to 24 hours in B2C configuration.
```